google-cloud-bigquery/lib/google/cloud/bigquery/dataset.rb

```diff
@@ -904,15 +910,16 @@ def query_job query, params: nil, external: nil,
                       legacy_sql: legacy_sql, standard_sql: standard_sql,
                       maximum_billing_tier: maximum_billing_tier,
                       maximum_bytes_billed: maximum_bytes_billed,
-                      params: params, external: external, labels: labels,
-                      udfs: udfs }
+                      job_id: job_id, prefix: prefix, params: params,
+                      external: external, labels: labels, udfs: udfs }
 
-          updater = QueryJob::Updater.from_options query, options
+          updater = QueryJob::Updater.from_options service, query, options
           updater.dataset = self
+          updater.location = location if location # may be dataset reference
```

Comment:
I know why you are using a conditional, but users may decide that they want to set the value to `nil`, so we should probably have `QueryJob::Updater#location=` do the right thing when setting the value to `nil`.

Reply:
What do you mean by "do the right thing"? Prevent a `nil`?